### PR TITLE
Support for SH1106 1.3" OLED display

### PR DESCRIPTION
**Pull request by phl0**
This adds support for a 1.3" OLED display connected to the I²C bus of the ESP8266.

### CODE REVIEW

File: src/bttrx_display.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+#ifdef ARDUINO
+
+#include "bttrx_display.h"
+#include "bttrx_logo.h"
+
+#include "SH1106Wire.h"
+#include "OLEDDisplayUi.h"
+
+extern BTTRX_DISPLAY bttrx_display;
```
Comment (chris007de) [inline src/bttrx_display.cpp:30]: Not necessary

File: src/bttrx_display.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+#ifdef ARDUINO
+
+#define display_width  128
+#define display_height  64
+
+#include <string>
+using namespace std;
+
+extern const char *GIT_REVISION;
```
Comment (chris007de) [inline src/bttrx_display.h:32]: Please include GIT_REVISION in bttrx_display.cpp as it is not used in the header, only in the cpp file

File: src/bttrx_logo.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+Copyright (C) 2019 Christian Obersteiner (DL1COM), Andreas Müller (DC1MIL)
+              2020 Florian Wolters (DF2ET)
+Contact: bt-trx.com, mail@bt-trx.com
+*/
+
```
Comment (chris007de) [inline src/bttrx_logo.h:21]: Add `#pragma once`

File: src/bttrx_fsm.h [hunk 4]
```diff
@@ -22,12 +22,14 @@ Contact: bt-trx.com, mail@bt-trx.com
 
 #ifdef ARDUINO
 #include <Arduino.h>
+#include "bttrx_display.h"
```
Comment (chris007de) [inline src/bttrx_fsm.h:25]: Duplicates the include in Line 32. This line can be removed, as bttrx_display.h has its own #ARDUINO include guard.

File: src/ptt.cpp [hunk 5]
```diff
@@ -92,6 +92,9 @@ void PTT::toggle(uint32_t delay_ms) {
     delayed_off(delay_ms);
   } else {
     on();
+#ifdef ARDUINO
+    bttrx_display_.setTransmitMessage("<<< ON AIR >>>");
```
Comment (chris007de) [inline src/ptt.cpp:96]: See PR comment

File: src/ptt.cpp [hunk 6]
```diff
@@ -105,6 +108,9 @@ void PTT::delayed_off(uint32_t delay_ms) {
 
   if (turn_off_delay_ == 0) {
     off();
+#ifdef ARDUINO
+    bttrx_display_.setTransmitMessage("idle");
```
Comment (chris007de) [inline src/ptt.cpp:112]: See PR comment

File: src/ptt.h [hunk 7]
```diff
@@ -22,6 +22,7 @@ Contact: bt-trx.com, mail@bt-trx.com
 
 #ifdef ARDUINO
 #include "Arduino.h"
+#include "bttrx_display.h"
```
Comment (chris007de) [inline src/ptt.h:25]: See PR comment

File: src/ptt.h [hunk 8]
```diff
@@ -41,6 +42,10 @@ class PTT {
   void toggle(uint32_t = 0);
   bool getState() { return ptt_on_; };
 
+#ifdef ARDUINO
+  BTTRX_DISPLAY bttrx_display_;
```
Comment (chris007de) [inline src/ptt.h:46]: See PR comment

File: src/main.cpp [hunk 9]
```diff
@@ -85,14 +88,12 @@ void setupPins() {
...
 
-String getHardwareVersion() {
+void getHardwareVersion() {
   switch (analogRead(PIN_HW_VER)) {
   case 0:
-    return "4.1";
+    hardwareVersion = "4.1";
   case 4095:
-    return "5.0";
-  default:
```
Comment (chris007de) [inline src/main.cpp:94]: Please add the default case again :fearful:

File: src/bttrx_display.cpp (hunk 1 above)
Comment (phl0) [reply src/bttrx_display.cpp:30]: Then includes are required. So I guess only line 30 ```extern BTTRX_DISPLAY bttrx_display;``` want ment here.

File: src/ptt.cpp (hunk 5 above)
Comment (phl0) [reply src/ptt.cpp]: Fixed with https://github.com/bt-trx/firmware/pull/46/commits/21f632558127e83ccd5750e958e7aaad47f23035.

File: src/ptt.h (hunk 7 above)
Comment (phl0) [reply src/ptt.h]: Fixed with https://github.com/bt-trx/firmware/pull/46/commits/21f632558127e83ccd5750e958e7aaad47f23035.

File: src/ptt.h (hunk 8 above)
Comment (phl0) [reply src/ptt.h]: Fixed with https://github.com/bt-trx/firmware/pull/46/commits/21f632558127e83ccd5750e958e7aaad47f23035.

**Pull request merged**